Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -154,82 +206,90 @@ private ContentTagTask(CentralRepository db, PropertyChangeEvent evt) {
-
-                    if (tags.stream()
-                            .map(tag -> tag.getName().getDisplayName())
-                            .filter(TagsManager.getNotableTagDisplayNames()::contains)
-                            .collect(Collectors.toList())
-                            .isEmpty()) {
-
-                        // There are no more bad tags on the object
-                        if (content instanceof AbstractFile) {
-                            af = (AbstractFile) content;
-                            knownStatus = TskData.FileKnown.UNKNOWN;
-                            comment = "";
-                        } else {
-                            LOGGER.log(Level.WARNING, "Error updating non-file object");
-                            return;
-                        }
-                    } else {
-                        // There's still at least one bad tag, so leave the known status as is
-                        return;
-                    }
-                } catch (TskCoreException | NoCurrentCaseException ex) {
-                    LOGGER.log(Level.SEVERE, "Failed to find content", ex);
-                    return;
+                if (hasNotableTag(tags)) {
+                    // There's still at least one bad tag, so leave the known status but update the comment to most recent
+                    String comment = getMostRecentNotableComment(tags);
+                    setContentKnownStatus(af, comment, TskData.FileKnown.BAD, false);
+                }
+                else {
+                    // There are no more bad tags on the object.  So, remove the comment if comment is same as CR tag
+                    String comment = tagDeletedEvent.getDeletedTagInfo().getComment();
+                    setContentKnownStatus(af, comment, TskData.FileKnown.UNKNOWN, true);
                 }
+            } catch (TskCoreException | NoCurrentCaseException ex) {
+                LOGGER.log(Level.SEVERE, "Failed to find content", ex);
+                return;
+            }
+        }
+        
+        private void handleTagAdded(ContentTagAddedEvent tagAddedEvent) {
+            // For added tags, we want to change the known status to BAD if the 
+            // tag that was just added is in the list of central repo tags.
+            final ContentTag tagAdded = tagAddedEvent.getAddedTag();
+            
+            // The added tag isn't flagged as bad in central repo, so do nothing
+            if (!isNotableTag(tagAdded))
+                return;
+            
+            AbstractFile af = getAbstractFile(tagAdded.getContent());
+            if (af == null)
+                return;
+            
+            setContentKnownStatus(af, tagAdded.getComment(), TskData.FileKnown.BAD, false);
+        }
+        
+        private AbstractFile getAbstractFile(Content content) {
+            if (content instanceof AbstractFile) {
+                return (AbstractFile) content;
+            } 
+            else {
```

Comment:
Our coding style for this project is to use end braces, i.e., "...} else {..." Are you remembering to format all files with the IDE before submitting them?